File: dart-wrapper/cardinal_sdk/lib/auth/captcha_options.dart
class CaptchaOptions {

  final Map<String, dynamic> _json;

  CaptchaOptions._internal(this._json);

  factory CaptchaOptions.Recaptcha(String solution) {
    return CaptchaOptions._internal({
      "solution": solution,
      "kotlinType": "com.icure.cardinal.sdk.dart.auth.CaptchaOptions.Recaptcha"
    });
  }

  factory CaptchaOptions.FriendlyCaptcha(String solution) {
    return CaptchaOptions._internal({
      "solution": solution,
      "kotlinType": "com.icure.cardinal.sdk.dart.auth.CaptchaOptions.FriendlyCaptcha"
    });
  }

  factory CaptchaOptions.KerberusComputed(Solution solution) {
    return CaptchaOptions._internal({
      "solution": Solution.encode(solution),
      "kotlinType": "com.icure.cardinal.sdk.dart.auth.CaptchaOptions.Kerberus.Computed"
    });
  }

  factory CaptchaOptions.KerberusDelegated() {
    return CaptchaOptions._internal({
      "kotlinType": "com.icure.cardinal.sdk.dart.auth.CaptchaOptions.Kerberus.Delegated"
    });
  }

  static Map<String, dynamic> encode(CaptchaOptions value) {
    return Map.unmodifiable(value._json);
  }
}

class Solution {
  final String id;
  final List<String> nonces;

  Solution(this.id, this.nonces);

  static Map<String, dynamic> encode(Solution value) {
    return {
      "id": value.id,
      "nonces": value.nonces
    };
  }

}

File: dart-wrapper/lib/src/commonMain/kotlin/com/icure/cardinal/sdk/dart/auth/CaptchaOptions.kt
package com.icure.cardinal.sdk.dart.auth

import com.icure.kerberus.Solution
import kotlinx.serialization.Serializable

@Serializable
sealed interface CaptchaOptions {

	fun toKt(): com.icure.cardinal.sdk.auth.CaptchaOptions

	@Serializable
	data class Recaptcha(val solution: String) : CaptchaOptions {
		override fun toKt() =
			com.icure.cardinal.sdk.auth.CaptchaOptions.Recaptcha(solution)
	}

	@Serializable
	data class FriendlyCaptcha(val solution: String) : CaptchaOptions {
		override fun toKt() =
			com.icure.cardinal.sdk.auth.CaptchaOptions.FriendlyCaptcha(solution)
	}

	@Serializable
	sealed interface Kerberus : CaptchaOptions {

		@Serializable
		data class Computed(val solution: Solution) : Kerberus {
			override fun toKt() =
				com.icure.cardinal.sdk.auth.CaptchaOptions.Kerberus.Computed(solution)
		}

		@Serializable
		data object Delegated : Kerberus {
			override fun toKt() =
				com.icure.cardinal.sdk.auth.CaptchaOptions.Kerberus.Delegated()
		}

	}
}